be/src/runtime/tablets_channel.cpp

```diff
@@ -129,9 +129,16 @@ Status TabletsChannel::close(int sender_id, int64_t backend_id, bool* finished,
         for (auto writer : need_wait_writers) {
             // close may return failed, but no need to handle it here.
             // tablet_vec will only contains success tablet, and then let FE judge it.
-            writer->close_wait(
-                    tablet_vec, tablet_errors,
+            Status st = writer->close_wait(
+                    tablet_vec,
                     (_broken_tablets.find(writer->tablet_id()) != _broken_tablets.end()));
+            if (!st.ok()) {
+#ifndef BE_TEST
+                PTabletError* tablet_error = tablet_errors->Add();
```

Comment:
change the code to a function. and also move 
```#ifndef BE_TEST
    if (!is_broken) {
        PTabletInfo* tablet_info = tablet_vec->Add();
        tablet_info->set_tablet_id(_tablet->tablet_id());
        tablet_info->set_schema_hash(_tablet->schema_hash());
    }
#endif
```

 to this funciton to remove unless #ifndef